### PR TITLE
Add example of using externref in assembly code

### PR DESCRIPTION
**Pull request by sbc100**
See #15878

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -8835,6 +8835,14 @@ def test_em_async_js(self):
     self.maybe_closure()
     self.do_core_test('test_em_async_js.c')
 
+  @require_v8
+  def test_externref(self):
+    self.run_process([EMCC, '-c', test_file('core/test_externref.s'), '-o', 'asm.o'])
+    self.emcc_args += ['--js-library', test_file('core/test_externref.js')]
+    self.emcc_args += ['-mreference-types']
+    self.node_args += ['--experimental-wasm-anyref']
```
Comment (kripken) [inline tests/test_core.py:8843]:
```suggestion
    self.v8_args += ['--experimental-wasm-anyref']
```
as the test runs in v8 and not node

Comment (sbc100) [reply tests/test_core.py]: Ah thanks.. turns out no argument is needed after all since recent d8 has this enabled by default.

**Review by kripken — APPROVED** [review]
Nice that this works so easily!

I wonder if maybe we should add something pointing to this test in the docs? Or maybe it's too early.

**Comment by sbc100** [comment]
> Nice that this works so easily!
> 
> I wonder if maybe we should add something pointing to this test in the docs? Or maybe it's too early.

Perhaps as a followup.   

I'm a little worried that we might still want to make changes to the `.s` format, so we might not want to be advertising it too loudly... although its been stable for a long time now.

**Comment by Thaina** [comment]
As it seem LLVM introduce support for externref in c, 

https://github.com/llvm/llvm-project/blob/main/clang/test/CodeGen/WebAssembly/wasm-externref.c

Can we have example to use externref in c too?

**Pull request merged**